integrated user login into create product

austinkwok1997 opened the pull request:
@bonnieli 

page now runs loadUser on startup

products are now assigned to proper user

each user will have their own directory in amazon s3 bin

austinkwok1997 committed:
integrated user login into create product

austinkwok1997 committed:
typo fix

bonnieli commented on client/src/pages/CreateProduct.js:
```diff
@@ -68,28 +69,35 @@ const typeOfProduct = [
+      secretAccessKey: process.env.REACT_APP_AWS_SECRET_KEY,
+      dirName: user._id,
+    }
+    S3Client = new S3(config);
```
is there a reason why you need to reassign `S3Client`? i feel like the way you had it before would be ok `const S3Client = new S3(config)`

bonnieli replied:
oh it might be because `user._id` may not be ready right? in that case, i may explore the option of using `useEffect` with a dependency on `user` context, if it changes it will update this config and S3Client :)

bonnieli commented on client/src/pages/CreateProduct.js:
```diff
@@ -68,28 +69,35 @@ const typeOfProduct = [
+  let S3Client = new S3();
+
+  useEffect(() => {
+    loadUser(localStorage.getItem("token"));
```
instead of loadingUser here, you may want to move this up the parent, maybe in `App.js` so that it just loads the user once every session

austinkwok1997 committed:
Moved general useEffect function to authState

added useEffect for changes in user in CreateProduct

disabled buttons until user has been loaded in Create Product

austinkwok1997 commented:
@bonnieli 
 
I moved general useEffect function to authState so that it should run on all pages.

added useEffect for changes in user in CreateProduct

disabled buttons until user has been loaded in Create Product
